cpp/models/ode_secirts/model.h

```diff
@@ -0,0 +1,971 @@
+    Flow<InfectionState::InfectedSymptomsImprovedImmunity,            InfectionState::InfectedSevereImprovedImmunity>,
+    Flow<InfectionState::InfectedSymptomsImprovedImmunity,            InfectionState::TemporaryImmunImprovedImmunity>,
+    Flow<InfectionState::InfectedSymptomsImprovedImmunityConfirmed,   InfectionState::InfectedSevereImprovedImmunity>,
+    Flow<InfectionState::InfectedSymptomsImprovedImmunityConfirmed,   InfectionState::TemporaryImmunImprovedImmunity>,
+    Flow<InfectionState::InfectedSevereImprovedImmunity,              InfectionState::InfectedCriticalImprovedImmunity>,
+    Flow<InfectionState::InfectedSevereImprovedImmunity,              InfectionState::TemporaryImmunImprovedImmunity>,
+    Flow<InfectionState::InfectedSevereImprovedImmunity,              InfectionState::DeadImprovedImmunity>,
+    Flow<InfectionState::InfectedCriticalImprovedImmunity,            InfectionState::DeadImprovedImmunity>,
+    Flow<InfectionState::InfectedCriticalImprovedImmunity,            InfectionState::TemporaryImmunImprovedImmunity>,
+    
+    // waning
+    Flow<InfectionState::TemporaryImmunPartialImmunity,                          InfectionState::SusceptiblePartialImmunity>,
+    Flow<InfectionState::TemporaryImmunImprovedImmunity,                          InfectionState::SusceptibleImprovedImmunity>,
+    Flow<InfectionState::SusceptibleImprovedImmunity,                 InfectionState::SusceptiblePartialImmunity>,
+    Flow<InfectionState::SusceptiblePartialImmunity,                  InfectionState::SusceptibleNaive>>;
+
+// clang-format on
+
+template <typename FP = ScalarType>
+class Model
+    : public FlowModel<FP, InfectionState, mio::Populations<FP, AgeGroup, InfectionState>, Parameters<FP>, Flows>
+{
+    using Base = FlowModel<FP, InfectionState, mio::Populations<FP, AgeGroup, InfectionState>, Parameters<FP>, Flows>;
+
+public:
+    using typename Base::ParameterSet;
+    using typename Base::Populations;
+
+    Model(const Populations& pop, const ParameterSet& params)
+        : Base(pop, params)
+    {
+    }
+
+    Model(int num_agegroups)
+        : Model(Populations({AgeGroup(num_agegroups), InfectionState::Count}), ParameterSet(AgeGroup(num_agegroups)))
+    {
+    }
+
+    void get_flows(Eigen::Ref<const Vector<FP>> pop, Eigen::Ref<const Vector<FP>> y, FP t,
+                   Eigen::Ref<Vector<FP>> flows) const override
+    {
+        auto const& params   = this->parameters;
+        AgeGroup n_agegroups = params.get_num_groups();
+
+        ContactMatrixGroup const& contact_matrix = params.template get<ContactPatterns<FP>>();
+
+        auto icu_occupancy           = 0.0;
+        auto test_and_trace_required = 0.0;
+        for (auto i = AgeGroup(0); i < n_agegroups; ++i) {
+            test_and_trace_required +=
+                (1 - params.template get<RecoveredPerInfectedNoSymptoms<FP>>()[i]) /
+                params.template get<TimeInfectedNoSymptoms<FP>>()[i] *
```

Comment:
Here is an error. If you want to use the capacity, then you should take into account the reduced time to recover for the more protected states. I think it is also wrong in Wadim's former model. Let's discuss on Monday.